backend/capellacollab/projects/capellamodels/routes.py

```diff
@@ -4,63 +4,149 @@
-):
-    project = projects_crud.get_slug(db, project_slug)
-    verify_project_role(project.name, token, db)
+@router.get("/", response_model=t.List[ResponseModel])
+def get_id(
+    project_slug: str, db: Session = Depends(get_db)
+) -> t.List[ResponseModel]:
+
+    project = projects_crud.get_project_by_slug(db, project_slug)
+    if not project:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The project having the name {project_slug} was not found.",
+                "technical": f"No project with {project_slug} found.",
+            },
+        )
     return [
         ResponseModel.from_model(model)
-        for model in crud.get_all(db, project_slug)
+        for model in crud.get_all_models(db, project.slug)
     ]
 
 
-@router.get("/{project_slug}/details/")
+@router.get("/{slug}", response_model=ResponseModel)
 def get_slug(
     project_slug: str,
-    id: t.Optional[int],
+    slug: str,
     db: Session = Depends(get_db),
-    token=Depends(JWTBearer()),
-):
-    project = projects_crud.get_slug(db, project_slug)
-    verify_project_role(project, token, db)
-    return ResponseModel.from_model(crud.get_slug(db, project_slug, id))
+) -> ResponseModel:
+
+    project = projects_crud.get_project_by_slug(db, project_slug)
+    if not project:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The project having the name {project_slug} was not found.",
+                "technical": f"No project with {project_slug} found.",
+            },
+        )
+    model = crud.get_model_by_slug(db, project_slug, slug)
+    if not model:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The model having the name {slug} of the project {project.name} was not found.",
+                "technical": f"No model with {slug} found in the project {project.name}.",
+            },
+        )
 
+    response_model = ResponseModel.from_model(model)
+    return response_model
 
-@router.post("/{project_slug}/create-empty", response_model=ResponseModel)
-def create_empty(
-    project_slug,
-    model: NewModel,
+
+@router.post("/", response_model=ResponseModel)
+def create_new(
+    project_slug: str,
+    new_model: CapellaModel,
     db: Session = Depends(get_db),
-    token=Depends(JWTBearer()),
+    token: JWTBearer = Depends(JWTBearer()),
 ) -> ResponseModel:
-    project = projects_crud.get_slug(db, project_slug)
+
+    project = projects_crud.get_project_by_slug(db, project_slug)
+    if not project:
+        raise HTTPException(
+            404,
+            {
+                "reason": f"The project having the name {project_slug} was not found.",
+                "technical": f"No project with {project_slug} found.",
+            },
+        )
     verify_project_role(
         repository=project.name,
         token=token,
         db=db,
         allowed_roles=["manager", "administrator"],
     )
```

Comment:
But we don’t know the project name before doing such a request, this implies that we need to change the verify_project_role function to take the slug as argument

Reply:
Also, I’m aware that there is anyway a problem in the fact that a 404 error is raised before a 403 and that it could be used by an attacker, but same, it would need to rework the `verify_project_role` function

Reply:
Indeed, the `verify_project_role` has to be reworked slightly. Instead of the repository parameter (which was the unique identifier of a project before), you can pass the slug (which is the new identifier). But you have to make sure that every occurrence is updated. I started the process here: https://github.com/DSD-DBS/capella-collab-manager/pull/229, but it's not finished yet - therefore it would be nice to see the change as part of this PR.

Reply:
I really think that this should be a PR on its own, I have 4 other PR that wait for this one, and it’s also already pretty big and overflows its scope by a lot. And also because it would be easier to change all it’s occurrences if it’s in one PR.

Reply:
And also, I think it could be solved way more elegantly. The solution would be to introduce a FastAPI injectable `User` object (not yet sure of the name), whose constructor returns a `User` or raises a 403.

Reply:
Tracked in https://github.com/DSD-DBS/capella-collab-manager/issues/247